logstash-core/lib/logstash/pipeline.rb

```diff
@@ -250,12 +250,32 @@ def stopped?
     @running.false?
   end
 
+  # register_plugin simply calls the plugin #register method and catches & logs any error
+  # @param plugin [Plugin] the plugin to register
+  # @return [Plugin] the registered plugin
+  def register_plugin(plugin)
+    plugin.register
+    plugin
+  rescue => e
+    @logger.error("Error registering plugin", :plugin => plugin.inspect, :error => e)
```

Comment:
I know this code snippet was my suggestion but let's avoid logging "raw" exception objects as we've had serialization issues in log4j2 in the past, maybe `:error => e.message` or backtrace